handlers/main.yml

```diff
@@ -1,6 +1,19 @@
 # file: postgresql/handlers/main.yml
 
-- name: restart postgresql
-  service:
-    name: "{{ postgresql_service_name }}"
-    state: restarted
+  - name: restart postgresql with service
+    service:
+      name: "{{ postgresql_service_name }}"
+      state: restarted
+      enabled: yes
+
+  - name: restart postgresql with systemd
+    systemd:
+      name: "{{ postgresql_service_name }}"
+      state: restarted
+      enabled: yes
+    when: (ansible_distribution == "CentOS" and ansible_distribution_major_version == "7") or
+          (ansible_distribution == "RedHat" and ansible_distribution_major_version == "7") or
+          (ansible_distribution == "Debian" and ansible_distribution_major_version == "8") or
+           ansible_os_family == "Ubuntu" and ansible_lsb.major_release|int >= 15.04 
```

Comment:
You can use the `ansible_service_mgr == 'systemd'` to check which init system is used, because an administrator can install another init even on a recent system (`sysvinit` on Debian Jessie LXC container is common)

Reply:
dang. I miss that. Alright let me update it